dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/EventExporterTest.java

```diff
@@ -592,23 +591,83 @@ void testExportEventsWhenFilteringByNumericDataElements()
   }
 
   @Test
-  void shouldFilterByEventsWithGivenDataValuesWhenFilterContainsDataElementUIDsOnly()
+  void shouldFilterByEventsContainingGivenDataValuesWhenFilteringByNonNullDataValues()
       throws ForbiddenException, BadRequestException {
     EventOperationParams params =
         EventOperationParams.builder()
+            .eventParams(EventParams.FALSE)
             .dataElementFilters(
                 Map.of(
                     UID.of("GieVkTxp4HH"),
-                    new ArrayList<>(),
```

Comment:
Did you decide not to improve the API of the EventOperationParams or will you do it later? Or is

`new ArrayList<>()` vs `List.of(new QueryFilter(QueryOperator.NNULL))`

not an issue anymore as we are dealing with it in the query params 🤔

Reply:
I still want to do it, but in a different PR